packages/forms/src/UIForm/fields/NestedListView/NestedListView.component.js

```diff
@@ -0,0 +1,254 @@
+import keycode from 'keycode';
+import ListView from '@talend/react-components/lib/ListView';
+import { translate } from 'react-i18next';
+
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import getDefaultT from '../../../translate';
+import { getItemsProps, initItems } from './NestedListView.utils';
+import FieldTemplate from '../FieldTemplate';
+
+import theme from './NestedListView.scss';
+
+const DISPLAY_MODE_DEFAULT = 'DISPLAY_MODE_DEFAULT';
+const DISPLAY_MODE_SEARCH = 'DISPLAY_MODE_SEARCH';
+const DEFAULT_ITEM_HEIGHT = 33;
+
+class NestedListViewWidget extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const { t, value, schema, id } = props;
+
+		this.defaultHeaderActions = [
+			{
+				id: `${id}-search`,
+				icon: 'talend-search',
+				label: t('LISTVIEW_WIDGET_SEARCH', { defaultValue: 'Search for specific values' }),
+				onClick: this.switchToSearchMode.bind(this),
+			},
+		];
+
+		this.defaultSearchHeaderActions = [
+			{
+				id: 'abort',
+				icon: 'talend-cross',
+				label: t('LISTVIEW_WIDGET_ABORT', { defaultValue: 'Abort' }),
+				onClick: this.switchToDefaultMode.bind(this),
+			},
+		];
+
+		const toggledChildren = [];
+
+		this.state = {
+			...initItems(schema, value, null, toggledChildren, this.getInitItemsCallbacks()),
+			toggledChildren,
+			value,
+			getItemHeight: () => DEFAULT_ITEM_HEIGHT,
+			headerDefault: this.defaultHeaderActions,
+			onAddKeyDown: this.onInputKeyDown.bind(this),
+			onInputChange: this.onInputChange.bind(this),
+		};
+	}
+
+	/**
+	 * Click callback to toggle the expanded status of children elements
+	 * @param {Object} event
+	 * @param {Object} item
+	 */
+	onExpandToggle(event, item) {
+		this.setState(state => {
+			const { items, value, searchCriteria, toggledChildren } = state;
+			const { toggleId } = item;
+
+			const newToggledChildren = toggledChildren.includes(toggleId)
+				? toggledChildren.filter(toggled => toggled !== toggleId)
+				: [...toggledChildren, toggleId];
+
+			return getItemsProps(items, value, searchCriteria, newToggledChildren);
+		});
+	}
+
+	/**
+	 * Handle checked item (parent) in the ListView change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Object } item
+	 */
+	onParentChange(event, item) {
+		this.setState(state => {
+			const { items, value, searchCriteria, toggledChildren } = state;
+			const { enum: availableOptions } = this.props.schema.schema.properties[item.key].items;
+
+			// Toggle all values
+			const itemValue = value[item.key] || [];
+			value[item.key] = itemValue.length === 0 ? availableOptions : [];
+
+			this.onChange(event, value);
+
+			return {
+				value,
+				...getItemsProps(items, value, searchCriteria, toggledChildren),
+			};
+		});
+	}
+
+	/**
+	 * Handle checked item (child) in the ListView change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Object } checked
+	 * @param { Object } parent
+	 */
+	onCheck(event, checked, parent) {
```

Comment:
`checked` could be `checkedItem` (if I understand) otherwise, we expect just a boolean